root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs

```diff
@@ -0,0 +1,222 @@
+        public AsyncProcessingServiceParameterValue UpdateServiceData(AsyncProcessingServiceParameterValue.AsyncStatus checkStatus, string userId, DateTime executionStartDateTime,
+                                                                      DateTime registrationDateTime, int numberOfRetries, int progressRate, DateTime completionDateTime,
+                                                                      int statusId, int commandId)
+        {
+            _asyncParameterValue = new AsyncProcessingServiceParameterValue("AsyncProcessingService", "Update", "Update", "SQL",
+                                                                             new MyUserInfo("AsyncProcessingService", "AsyncProcessingService"));
+            _asyncParameterValue.UserId = userId;
+            _asyncParameterValue.ExecutionStartDateTime = executionStartDateTime;
+            _asyncParameterValue.RegistrationDateTime = registrationDateTime;
+
+            // Based on the service status setting database command value
+            switch (checkStatus)
+            {
+                case AsyncProcessingServiceParameterValue.AsyncStatus.Processing:
+                    _asyncParameterValue.ProgressRate = 50;
+                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.Processing;
+                    _asyncParameterValue.CommandId = (int)AsyncProcessingServiceParameterValue.AsyncCommand.Stop;
+                    break;
+                case AsyncProcessingServiceParameterValue.AsyncStatus.Stop:
+                    _asyncParameterValue.ProgressRate = 100;
+                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.Stop;
+                    _asyncParameterValue.CommandId = 0;
+                    break;
+                case AsyncProcessingServiceParameterValue.AsyncStatus.AbnormalEnd:
+                    _asyncParameterValue.ProgressRate = 0;
+                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.AbnormalEnd;
+                    _asyncParameterValue.CommandId = 0;
+                    break;
+                case AsyncProcessingServiceParameterValue.AsyncStatus.Abort:
+                    _asyncParameterValue.ProgressRate = 0;
+                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.Abort;
+                    _asyncParameterValue.CommandId = 0;
+                    break;
+            }
+            _asyncParameterValue.NumberOfRetries = numberOfRetries;
+            _asyncParameterValue.CompletionDateTime = completionDateTime;
+
+            return _asyncParameterValue;
+        }
+
+        /// <summary>
+        /// Set things in motion so your service can do its work.
+        /// </summary>
+        protected override void OnStart(string[] args)
+        {
+            StartByThreadPool(AsyncProcessingServiceParameterValue.AsyncStatus.Processing);
```

Comment:
Is not assumed to be used as follows.
https://github.com/SymphonyTeleca/CloudOS/pull/2/files#diff-974cc62e7721855f32e1a1eda94af2c6R580